docs/dev/webapp/components.md

```diff
@@ -0,0 +1,98 @@
+You can read more about routing [here](https://angular.io/guide/router) in the Angular docs.
+
+## csar-sidebar 
+
+![](img/sidebar.png)
+
+The sidebar shows all available CSARs if you click on a CSAR it also shows its sub items.
+
+The sidebar component also manages CSAR upload it checks if the CSAR name is valid and if the User submits it uploads the file to the server.
+
+### csar-item 
+
+This component holds a CSAR and its sub items.
+
+It handles user actions like:
+
+- clicking on the csar-item: notify the RouteHandler to open the csar-view, highlight itself and show the csar-sub-item component
+- clicking the `+` button: notify the RouteHandler to open the transformation-creator component to create a transformation for the given csar
+- clicking the deletion button: delete the CSAR and close its csar-view if it is opened
+
+
+### csar-sub-item
+
+This component shows t list of transformations based on the given csar.
+
+It handles user actions like:
+
+- clicking on a transformations list item: notify the RouteHandler to open the transformation-view for transformation clicked on
+- clicking on the trash bin: delete the transformation and close the transformation-view if opened
+
+## transformation-creator
+
+![](img/transformation-creator.png)
+
+The transformation creator shows all available target platforms.
+If a user click on a target platform and a transformation for that platform already exists it shows a dialog that asks the user if he wants to overwrite the transformation.
+Then it creates a transformation for that platform and tells the RouteHandler to show the transformation-input view.
+
+## transformation-inputs
+
+![](img/transformation-inputs.png)
+
+This component shows a table containing all inputs a transformation needs.
+If a user fills one of the text boxes the component is responsible for validating the input. 
+
+## transformation-view
+
+![](img/view_transformation.png)
+
+This component shows all relevant information about a transformation.
+For each of the transformations phase it shows the state in a corresponding color.
+Also it creates the run script that automatically downloads the artifact of a transformation and executes the `deploy.sh`.
+The transformation-view also polls for changes in the transformation process and updates the log view it holds.
+
+Additionally to the log tab the component has also a inputs and outputs tab.
+The inputs tab shows the transformation inputs and the outputs tab shows the outputs that were defined in the service template of the CSAR.
+
+## csar-view
+
+![](img/view_csars.png)
+
+The csar-view shows information about a CSAR.
+Like the transformation-view it also shows the CSAR processing phases and its status.
+And it also holds the log view.
+
+## log
+
+![](img/log_view.png)
+
+The log shows the server log in a table.
+Angular is destroying the log formatting there fore the log component replaces all white spaces in a log entry with `&nbsp;`, a non-breaking space.
```

Comment:
`there fore` -> `therefore`

Reply:
You did address the whitespace issue - either say that this issue is resolved or don't talk about it here (I'd say this is what inline doc is for)